### DISCUSSION TITLE
Iterate range

### DISCUSSION THREAD

**Discussion by einord** [dotnet/csharplang#4155] [Language Ideas]
Currently there is no simple and clean way to iterate some code a set number of times.
This is useful when generating random strings, or creating data for testing and other things that should be done any set number of times.

Currently this can be done in multiple ways (each creates messy and unnecessarily harder code to read):
```csharp
// Classic for-loop
for (int i = 0; i < 10; i++)
{
    // ...
}

// Another way, but not intuitive
foreach (int value in Enumerable.Range(0, 10))
{
    // ...
}
```

Since C# 8 it is possible to use the new range syntax with arrays (`myArray[1..10]`), and this similar syntax could easily be used for foreach iterations:

```csharp
foreach (int i in 0..10)
{
    // ...
}
```

This would make the code a lot cleaner, very easy to understand and perhaps this could also be used in combination with discards if the current range value is not important:
```csharp
foreach (_ in 0..10)
{
    // ...
}
```

Everything is based on previously existing syntax and feels intuitive in the C# language.

**Comment by CyrusNajmabadi** [comment]
> and this similar syntax could easily be used for foreach iterations:

This is already possible today. You just need an extension GetEnumerator method for System.Range

**Comment by jnm2** [comment]
See https://github.com/dotnet/csharplang/discussions/3004 for existing discussion on this, too.
